### PR TITLE
MONGOCRYPT-578: allowlist search index commands for auto encryption

### PR DESCRIPTION
**Pull request by baileympearson**

### CODE REVIEW

File: test/test-mongocrypt-ctx-encrypt.c [hunk 1]
```diff
@@ -959,6 +959,9 @@ static void _test_encrypt_init_each_cmd(_mongocrypt_tester_t *tester) {
     _init_bypass(tester, "{'getCmdLineOpts': 1}");
     _init_bypass(tester, "{'getLog': 1}");
     _init_ok(tester, "{'collMod': 'coll'}");
+    _init_bypass(tester, "{'createSearchIndex': 'coll' }");
+    _init_bypass(tester, "{'dropSearchIndex': 'coll' }");
+    _init_bypass(tester, "{'updateSearchIndex': 'coll' }");
```
Comment (addaleax) [inline test/test-mongocrypt-ctx-encrypt.c:964]: I’d add the `listSearchIndexes` command as well, unless we feel very confident that our plans to use and backport the agg stage are not going to change again

**Review by addaleax — APPROVED** [review]

File: src/mongocrypt-ctx-encrypt.c [hunk 2]
```diff
@@ -2471,6 +2471,14 @@ _check_cmd_for_auto_encrypt(mongocrypt_binary_t *cmd, bool *bypass, char **colln
         *bypass = true;
     } else if (0 == strcmp(cmd_name, "getLog")) {
         *bypass = true;
+    } else if (0 == strcmp(cmd_name, "createSearchIndex")) {
```
Comment (kevinAlbs) [inline src/mongocrypt-ctx-encrypt.c:2474]:
```suggestion
    } else if (0 == strcmp(cmd_name, "createSearchIndexes")) {
```

To follow naming described in DRIVERS-2549.

File: test/test-mongocrypt-ctx-encrypt.c [hunk 3]
```diff
@@ -959,6 +959,10 @@ static void _test_encrypt_init_each_cmd(_mongocrypt_tester_t *tester) {
     _init_bypass(tester, "{'getCmdLineOpts': 1}");
     _init_bypass(tester, "{'getLog': 1}");
     _init_ok(tester, "{'collMod': 'coll'}");
+    _init_bypass(tester, "{'listSearchIndexes': 'coll' }");
+    _init_bypass(tester, "{'createSearchIndex': 'coll' }");
```
Comment (kevinAlbs) [inline test/test-mongocrypt-ctx-encrypt.c:963]:
```suggestion
    _init_bypass(tester, "{'createSearchIndexes': 'coll' }");
```

**Review by kevinAlbs — APPROVED** [review]
LGTM with a possible typo fix. Thank you for the fix!

**Review by durran — CHANGES_REQUESTED** [review]
Just the requested `createSearchIndexes` typo changes.

**Pull request merged**